interface/src/app/greenelab/stats.js

```diff
@@ -0,0 +1,50 @@
+angular.module('greenelab.stats', [
+
+])
+
+.factory('MathFuncts', [function() {
+  return {
+    /*
+     * The binomLog and hyperGeometricTest functions are based heavily on the
+     * 'binomlog' and 'hyperg' functions in /integrator/static/js/imp.js
+     * See code here:
+     * https://bitbucket.org/cgreene/integrator/src/1194f7ae1b11057a4649114e1f55570542acbcde/integrator/static/js/imp.js?at=giant&fileviewer=file-view-default
+    */
+    binomLog: function(n, k) {
+      // The 'n' and 'k' arguments are the same as the 'n' and 'k' arguments
+      // arguments for the hyperGeometricTest function below (the number of
+      // items in the sample and the number of successes in the sample,
+      // respectively).
+      var coeff = 0;
+      for (var i = n - k + 1; i <= n; i++) {
+        coeff += Math.log(i);
+      }
+      for (i = 1; i <= k; i++) {
+        coeff -= Math.log(i);
+      }
+      return coeff;
+    },
+
+    hyperGeometricTest: function(k, m, n, N) {
+      // k is the number of successes in the sample
+      // m is the number of successes in the population
+      // n is the number of items in the sample
+      // N is the number of items in the population
+      //
+      // The names of these variables are chosen based partly on the variable
+      // names in the /integrator/static/js/imp.js script mentioned above,
+      // but are also common in online descriptions of the hypergeometric test.
+      // See: https://en.wikipedia.org/wiki/Hypergeometric_distribution#Application_and_example
+      var sum = 0;
+      for (var i = 0; i < k; i++) {
+        var logSum = this.binomLog(m, i) +
+            this.binomLog(N - m, n - i) - this.binomLog(N, n);
+        sum += Math.exp(logSum);
+      }
+      return sum;
+    }
+
+  };
+}])
```

Comment:
I haven't dug into the math itself to verify, but this is very nice and clear coding. Thank you for the excellent comments and organization!